### PR TITLE
implement the force 1 loop trip option

### PR DESCRIPTION
**Pull request by schweitzpgi**

### CODE REVIEW

File: flang/lib/Optimizer/Transforms/RewriteLoop.cpp [hunk 1]
```diff
@@ -22,6 +22,12 @@ static llvm::cl::opt<bool>
                          llvm::cl::desc("disable FIR to CFG pass"),
                          llvm::cl::init(false));
 
+/// minimum trip count is 1, not 0
+static llvm::cl::opt<bool> forceLoopToExecuteOnce(
```
Comment (vdonaldson) [inline flang/lib/Optimizer/Transforms/RewriteLoop.cpp:26]: Someone might expect this to also apply to unstructured loops.

File: flang/lib/Optimizer/Transforms/RewriteLoop.cpp [hunk 2]
```diff
@@ -68,22 +74,24 @@ class CfgLoopConv : public mlir::OpRewritePattern<fir::LoopOp> {
     rewriter.setInsertionPointToEnd(initBlock);
     auto diff = rewriter.create<mlir::SubIOp>(loc, high, low);
     auto distance = rewriter.create<mlir::AddIOp>(loc, diff, step);
-    auto iters = rewriter.create<mlir::SignedDivIOp>(loc, distance, step);
+    mlir::Value iters =
```
Comment (vdonaldson) [inline flang/lib/Optimizer/Transforms/RewriteLoop.cpp:77]: Is this auto --> mlir::Value change intentional?

Comment (schweitzpgi) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:77]: Yes.

File: flang/lib/Optimizer/Transforms/RewriteLoop.cpp (hunk 1 above)
Comment (schweitzpgi) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:26]: I can write an issue.

File: flang/lib/Optimizer/Transforms/RewriteLoop.cpp (hunk 2 above)
Comment (vdonaldson) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:77]: It seems out of character.  Is it required?

Comment (schweitzpgi) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:77]: Yes. We need to rewrite `iters` to be the value of a different operation when the option is thrown.

Comment (vdonaldson) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:77]: ok; thanks for the explanation

File: flang/lib/Optimizer/Transforms/RewriteLoop.cpp (hunk 1 above)
Comment (schweitzpgi) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:26]: https://github.com/flang-compiler/f18-llvm-project/issues/214

Comment (Sameeranjoshi) [reply flang/lib/Optimizer/Transforms/RewriteLoop.cpp:26]: #239

**Pull request merged**